### ISSUE TITLE
Compression into a smaller string ?

### ISSUE THREAD

**Issue by andyscott12** [meleze/Meleze.Web#2]
Hi :)

Cool work with this plugin - however, Google Closure is telling me that more efficiency can be achieved ? 

1) Is it possible to provide the ability to comment out certain things to "avoid" compression ? i.e. 

`<melezeavoid>blah blah blah<melezeavoid>`

 or something like this?

2) I think more compression can be achieved by joining the strings together to reduce whitespace between the lines ? i.e. check out 

view-source:https://trello.com/

And you'll see what I mean ?

**Comment by meleze** [comment]
For 1), that's a good idea. I would prefer a solution with HTML comments rather than a custom element to keep the page working even when the minimizer is not enabled.

For 2), I already tried to join the lines but this has side effects. The browsers don't interpret the whitespace between elements in the same way. This also change the DOM (and javascript) behavior.
This breaks applications like Orchard for example.

What we have is a "safe" setting. We could had an aggressive optimization mode for those that can fix their code to work in both normal and minized versions.

**Comment by meleze** [comment]
Something more...

For now, if I remember correctly, the javascript code is not changed all. 
To optimize a bit more your pages, you can externalize your javascript in dedicated files and minify them with JS specific libraries.

A possible enhancement could be to integrate such a library in the parser for the inline javascript.

**Comment by andyscott12** [comment]
@meleze 

yeah agree that 1) would be better with html comments - i.e. you could add some command specific to avoid comments

2) I think - perhaps like Google Closure - "simple" and "Advanced" settings could be good ? Allows applications that can be more aggressive to get even more performance gains ? i.e. like the minification within "view-source:https://trello.com/" would be "Advanced" ?

3) I don't think that javascript should matter ? i.e. its basically just removing all whitespace on the page [or as much as possible]

**Comment by andyscott12** [comment]
Any chance to try to update this so it's in a smaller string ?

And maybe allow html comments ? :)

**Comment by meleze** [comment]
I am going to release a new version soon (this WE probably) with an "aggressive" mode and stuff for comments and javascript.

**Comment by andyscott12** [comment]
awesome! you rock! looking forward to it :)

**Comment by andyscott12** [comment]
so tried the latest update :) it's generally great

however the "Aggressive" breaks a lot of inline JS on pages. Is there anyway to include a "preserve" option or just minify with the simple version and remove "new lines" only as a "Basic" or "intermediate" option - this might not be so "aggressive" in breaking existing CSS styles or inline JS events ?

**Comment by meleze** [comment]
You can still disable the aggressive mode by adding 
<add key="meleze-minifier:Aggressive" value="false"/>
in the appSettings of the main web.config. This will use the safe minifier (that has some optimizations compare to the previous release)

If you can send me examples of code broken by the minifier, I will try to fix the "Aggressive" mode :-)

**Comment by andyscott12** [comment]
1) It seems inline javascript written across multiple lines has a problem i.e. like

```
var myClass = 
someClass.something.test(function() {
//something
}).css({ color:'#333' });
```

It seems to return `missing } after function body`. I've noticed when looking at the code - that it seems to  return

```
var myClass = someClass.something.test(function() { //something }) .css({ color:'#333' });
```

Notice the "space" after the `}) .css` - it has done this a lot at least to me. Which seems to generate a lot of errors in "Aggressive" mode ?

2) URL's via script seem to have issues like

`$.getScript('https://www.test.com/test/test.js');`

Turns out in "aggressive" like

`$.getScript('https:` which generates a `unterminated string literal` ? Perhaps because of the "double" `//` ?
